### PR TITLE
[Process] Fix write access check for pipes on Windows

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19336, #19416 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Process/Pipes/WindowsPipes.php [hunk 1]
```diff
@@ -52,27 +52,30 @@ public function __construct($disableOutput, $input)
...
                     $file = sprintf('%s\\sf_proc_%02X.%s', $tmpDir, $i, $name);
-                    if (file_exists($file) && !@unlink($file)) {
+                    if (file_exists($file) && !unlink($file)) {
                         continue 2;
                     }
-                    $h = @fopen($file, 'xb');
+                    $h = fopen($file, 'xb');
+                    if (!$h && false === strpos($error, 'File exists')) {
+                        restore_error_handler();
+                        throw new RuntimeException('A temporary file could not be opened to write the process output to, verify that your TEMP environment variable is writable');
```
Comment (iltar) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: What about this?
```
A temporary file could not be opened to write the process output to. Verify that your TEMP environment variable is writable.
```

I've changed it to a full stop before Verify and added one on the end for consistency with other exceptions.

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: we don't change exception messages in minor versions...

Comment (fabpot) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: Yes wé can

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: If wé can, then I'd even prefer: `sprintf(''A temporary file could not be opened to write the process output: %s', $error)`

Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Pipes/WindowsPipes.php]: fixed

**Comment by fernandobandeira** [comment]
Thank you for this quick PR.
I've done more tests and had no more problems since this PR,
Before the exception were killing the process, now it's just a warning.

I were waiting for the author of #19336 to test his issue but it's been over a week, so for me at least this is Status:Reviewed

**Comment by nicolas-grekas** [comment]
Thanks. The warning should be gone btw.

**Pull request merged**